company-lsp.el

```diff
@@ -235,27 +232,31 @@ CANDIDATE is a string returned by `company-lsp--make-candidate'."
          (additional-text-edits (gethash "additionalTextEdits" item)))
     (cond
      (text-edit
-      (goto-char (line-beginning-position))
-      (delete-char (- (line-end-position) (line-beginning-position)))
-      (insert company-lsp--line-backup)
-      (lsp--apply-text-edit text-edit)
+      (setq insert-text (gethash "newText" text-edit))
+      (delete-region (- (point) (length candidate)) (point))
       (let* ((range (gethash "range" text-edit))
              (start-point (lsp--position-to-point (gethash "start" range)))
-             (new-text-length (length (gethash "newText" text-edit))))
+             (new-text-length (length insert-text)))
+        ;; set end of the range to be equal to the start of the range
+        ;; since we have already deleted the prefix.
+        (puthash "end" (gethash "start" range) range)
+        (lsp--apply-text-edit text-edit)
         (goto-char (+ start-point new-text-length))))
      ((and insert-text (not (eq insert-text-format 2)))
       (cl-assert (string-equal (buffer-substring-no-properties start (point)) label))
       (goto-char start)
       (delete-char (length label))
       (insert insert-text)))
-    (when additional-text-edits
-      (lsp--apply-text-edits additional-text-edits))
-    (when (and company-lsp-enable-snippet
-               (fboundp 'yas-expand-snippet))
-      (if (and insert-text (eq insert-text-format 2))
-          (yas-expand-snippet insert-text start (point))
-        (-when-let (fallback-snippet (company-lsp--fallback-snippet item))
-          (yas-expand-snippet fallback-snippet))))
+
+    (let ((start-marker (set-marker (make-marker) start)))
```

Comment:
Should clean up the marker at the end of the `let`:

```elisp
(set-marker start-marker nil)
```

Reply:
Thanks, sure! I was using markers for the first time :)